recipes/default.rb

```diff
@@ -0,0 +1,36 @@
+include_recipe 'apt'
+
+type = case node['platform_family']
+       when 'fedora', 'rhel'
+         'rpm'
+       else
+         'deb'
+       end
+
+packagecloud_repo 'StackStorm/stable' do
+  type type
+end
+
+package 'st2mistral'
+
+# ensure directories are managed
+[
+  node['openstack-mistral']['etc_dir']
+].each do |path|
+  directory "creating mistral directory #{path}" do
+    path path
+    owner 'root'
+    group 'root'
+    mode '0755'
+    action :create
+  end
+end
+
+template '/etc/mistral/mistral.conf' do
+  source 'mistral.conf.erb'
+  variables(
+    options: node['openstack-mistral']['config']
+  )
+  mode '0644'
+  action :create
+end
```

Comment:
reload would probably be better than restart (if allowed) and an attribute might want to be added to control if this happens or not.  Some people don't like their services being restarted on config changes and prefer to do them manually (db's are the main case this)